# keys
uint32 entity_uid
uint32 log_uid

# who
uint32[] episode_uids

# when
time timestamp

# what
string[] new_f
string[] updated_f
string[] removed_f